packages/nimble-components/src/table-column/text/tests/table-column-text.spec.ts

```diff
@@ -0,0 +1,197 @@
+import { html } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import type { Table } from '../../../table';
+import { TableColumnText } from '..';
+import { waitForUpdatesAsync } from '../../../testing/async-helpers';
+import { type Fixture, fixture } from '../../../utilities/tests/fixture';
+import type { TableRecord } from '../../../table/types';
+import { TablePageObject } from '../../../table/tests/table.pageobject';
+
+interface SimpleTableRecord extends TableRecord {
+    field?: string | null;
+    noPlaceholder?: string | null;
+    anotherField?: string | null;
+}
+
+const simpleTableData = [
+    {
+        field: 'string 1',
+        noPlaceholder: 'string 2',
+        anotherField: 'foo'
+    },
+    {
+        field: '<button></button>',
+        noPlaceholder: 'string 2',
+        anotherField: 'foo'
+    },
+    {
+        field: '',
+        noPlaceholder: 'string 2',
+        anotherField: 'foo'
+    },
+    {
+        field: null,
+        noPlaceholder: null,
+        anotherField: 'foo'
+    },
+    {
+        field: undefined,
+        noPlaceholder: undefined,
+        anotherField: 'foo'
+    }
+] as const;
+
+const tableColumnText = DesignSystem.tagFor(TableColumnText);
+
+// prettier-ignore
+async function setup(): Promise<Fixture<Table<SimpleTableRecord>>> {
+    return fixture<Table<SimpleTableRecord>>(
+        html`<nimble-table>
+                <${tableColumnText} field-name="field" placeholder="no value">
+                    Column 1
+                </${tableColumnText}>
+                <${tableColumnText} field-name="noPlaceholder">
+                    Column 2
+                </${tableColumnText}>
+            </nimble-table>`
+    );
+}
+
+describe('TableColumnText', () => {
+    let element: Table<SimpleTableRecord>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TablePageObject<SimpleTableRecord>;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        pageObject = new TablePageObject<SimpleTableRecord>(element);
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('displays placeholder string when no value present', async () => {
+        element.data = [...simpleTableData];
```

Comment:
Maybe it's just me, but I find these tests to be a little difficult to follow because you have to look back at the values in `simpleTableData` for these assertions to make sense. What about writing the tests to have the data local in the tests since each test is using different indexes in the data anyway? I'm thinking something like this:

```TS
element.data = [{ field: 'foo', noPlaceholder: 'bar', anotherField: 'baz' }];
await connect();
await waitForUpdatesAsync();

expect(getRenderedCellContent()).toBe('foo');
expect(getRenderedCellContent()).toBe('bar');
```

I think it could also help the tests from being brittle because changing `simpleTableData` won't affect every test in this file.

Reply:
Way better. Thanks! Note that I've opted to just use a single column for these tests, as having more than one column for what we're testing seems unnecessary. Multi-column verification happens in the table tests.